pkg/controller/kogitotrusty/kogitotrusty_controller.go

```diff
@@ -0,0 +1,161 @@
+			AddToScheme:  routev1.Install,
+			Objects:      []runtime.Object{&routev1.Route{}},
+		},
+		{
+			GroupVersion: imgv1.GroupVersion,
+			AddToScheme:  imgv1.Install,
+			Objects:      []runtime.Object{&imgv1.ImageStream{}},
+		},
+		{Objects: []runtime.Object{&corev1.Service{}, &appsv1.Deployment{}}},
+	}
+	controllerWatcher := framework.NewControllerWatcher(r.(*ReconcileKogitoTrusty).client, mgr, c, &appv1alpha1.KogitoTrusty{})
+	if err = controllerWatcher.Watch(watchedObjects...); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileKogitoTrusty implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileKogitoTrusty{}
+
+// ReconcileKogitoTrusty reconciles a KogitoTrusty object
+type ReconcileKogitoTrusty struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client *client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a KogitoTrusty object and makes changes based on the state read
+// and what is in the KogitoTrusty.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileKogitoTrusty) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.With("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling KogitoTrusty")
+
+	reqLogger.Infof("Injecting Trusty Index URL into KogitoApps in the namespace '%s'", request.Namespace)
+	if err := infrastructure.InjectTrustyURLIntoKogitoApps(r.client, request.Namespace); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	instances := &appv1alpha1.KogitoTrustyList{}
+	definition := services.ServiceDefinition{
+		DefaultImageName:    infrastructure.DefaultTrustyImageName,
+		Request:             request,
+		OnDeploymentCreate:  r.onDeploymentCreate,
+		KafkaTopics:         kafkaTopics,
+		RequiresPersistence: true,
+		RequiresMessaging:   true,
+		HealthCheckProbe:    services.QuarkusHealthCheckProbe,
+	}
+	if requeueAfter, err := services.NewSingletonServiceDeployer(definition, instances, r.client, r.scheme).Deploy(); err != nil {
+		return reconcile.Result{}, err
+	} else if requeueAfter > 0 {
+		return reconcile.Result{RequeueAfter: requeueAfter, Requeue: true}, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+const (
+	// Collection of kafka topics that should be handled by the Trusty
+	kafkaTopicNameTraceEvents string = "kogito-tracing-decision"
+)
+
+var kafkaTopics = []services.KafkaTopicDefinition{
+	{TopicName: kafkaTopicNameTraceEvents, MessagingType: services.KafkaTopicIncoming},
+}
+
+func (r *ReconcileKogitoTrusty) onDeploymentCreate(deployment *appsv1.Deployment, kogitoService appv1alpha1.KogitoService) error {
+	return nil
```

Comment:
Here's. It's not doing anything. We can just remove this function.

Reply:
Right, got it thanks!